### PR TITLE
[BitSail]refactor and enhance fake connector to support more Data-Generator

### LINKED ISSUE #148: [BitSail][Connector] FakeSource supports generating unique columns.
## Is your feature request related to a problem? Please describe

Current `FakeSource` supports generate unique long value.
But sometimes we may want to generate more type of unique columns, like string  or date.

This issue wants to support generate unique columns for the following type:
 - int
 - short
 - string
 - bigint
 - date

### PR DESCRIPTION
**Pull request by laglangyue**
Signed-off-by:

## Pre-Checklist

Note: Please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/bytedance/bitsail/blob/master/docs/contributing.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests.

## Purpose
According to Issue#148, enhance to supported unique data.
At present, all data generation is concentrated in one class, which is full of judgments and responsibilities. I refactored the fake data, so that more data generators can be easily extended.
And the unique data can be realized with Auto-Increament-RowID . 
The next step,I want to implement random data with unique characteristics. The current idea is to generate random data and 'closed hashing`.

## Approaches
1. refactor fakeData so that  more data generators can be easily extended.
2. generator unique data by Auot-Increament-RowID
3. generator unique data by random
4. add docment about fake-data

## Related Issues

Close #148 

## New Behavior (screenshots if needed)

N/A

### CODE REVIEW

**Comment by laglangyue** [comment]
I submit this PR to compare my code with master，I didn't complete the Step3 and Step4.

File: bitsail-connectors/connector-fake/src/main/java/com/bytedance/bitsail/connector/fake/source/generate/GeneratorMatcher.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    // Data generator by snowFlake
+    if (CollectionUtils.isNotEmpty(typeInfo.getTypeProperties()) && typeInfo.getTypeProperties().contains(TypeProperty.UNIQUE)) {
+      if (TypeInfos.LONG_TYPE_INFO.getTypeClass() == typeInfo.getTypeClass()) {
+        return new SnowflakeId(generateConfig.getTaskId());
+      }
+      // Data Generator by AutoIncrementData
+      return new AutoIncrementData(typeInfo);
+    }
+    // Data Generator by FakeData
+    if (TypeInfos.LONG_TYPE_INFO.getTypeClass() == typeInfo.getTypeClass()) {
```
[the remark this answers is not in the document]
Comment (laglangyue) [reply bitsail-connectors/connector-fake/src/main/java/com/bytedance/bitsail/connector/fake/source/generate/GeneratorMatcher.java]: done

**Comment by laglangyue** [comment]
> @laglangyue Some UT cases and Integration Tests failed, plz take a look.
> 
> > Caused by: java.lang.RuntimeException: Unsupported type LocalDate
> > at com.bytedance.bitsail.connector.fake.source.generate.GeneratorMatcher.match(GeneratorMatcher.java:50)

I don't know how fix the exception, can I run Integration test locally and how?.

**Review by healchow — APPROVED** [review]

File: bitsail-connectors/connector-fake/src/main/java/com/bytedance/bitsail/connector/fake/source/generate/GeneratorMatcher.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          match(((MapTypeInfo<?, ?>) typeInfo).getValueTypeInfo(), generateContext));
+    }
+
+    // Data generator by snowFlake
+    if (supportedUnique(typeInfo)) {
+      if (TypeInfos.LONG_TYPE_INFO.getTypeClass() == typeInfo.getTypeClass()) {
+        return new SnowflakeId(generateContext.getTaskId());
+      }
+      // Data Generator by AutoIncrementData
+      return new AutoIncrementData(typeInfo);
```
[the remark this answers is not in the document]
Comment (laglangyue) [reply bitsail-connectors/connector-fake/src/main/java/com/bytedance/bitsail/connector/fake/source/generate/GeneratorMatcher.java:42]: Yes, this is more accurate

**Pull request closed without merging**